src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -237,7 +237,7 @@ export const useSwapper = () => {
       case 'eip155:1':
         {
           const approvalFee = bnOrZero(trade?.feeData?.chainSpecific?.approvalFee)
-            .dividedBy(bn(10).exponentiatedBy(18))
+            .dividedBy(bn(10).exponentiatedBy(feeAsset.precision))
```

Comment:
Good find.

Here's a stylistic suggestion, as it tends to be a bit cleaner.

```suggestion
            .dividedBy(`1e+${feeAsset.precision}`))
```